packages/react-components/react-persona/Spec.md

```diff
@@ -1,63 +1,279 @@
+
+### Text lines
+
+Other than styling and naming, the text lines in Persona remain the same.
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+Persona with Avatar:
+
+```jsx
+<Persona primaryText="Kevin Sturgis" secondaryText="Software Engineer" avatar={{ name: 'Kevin Sturgis' }} />
+```
+
+Persona with Avatar + PresenceBadge:
+
+```jsx
+<Persona
+  primaryText="Kevin Sturgis"
+  secondaryText="Software Engineer"
+  tertiaryText="Offline"
+  avatar={{ name: 'Kevin Sturgis' }}
+  badge={{ status: 'offline', outOfOffice: true }}
+/>
+```
+
+Persona with icon:
+
+```jsx
+<Persona primaryText="Person Call Icon" icon={<PersonCallRegular />} />
+```
+
+Persona with PresenceBadge:
+
+```jsx
+<Persona primaryText="Kevin Sturgis" badge={{ status: 'offline', outOfOffice: true }} />
+```
+
+Persona with image:
+
+```jsx
+<Persona
+  primaryText="Katri Athokas"
+  secondaryText="Software Engineer"
+  tertiaryText="CXE"
+  quaternaryText="Offline"
+  image={{ src: 'https://fabricweb.azureedge.net/fabric-website/assets/images/avatar/KatriAthokas.jpg' }}
+/>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+There are three alignment variants:
+
+- start: Coin on the left and text on the right.
+- center: Coin on top and text on the bottom.
+- end: Coin on the right and text on the left.
+
+There are 5 coin variats:
+
+- Badge
+- icon
+- Avatar
+- Avatar + Badge
+  - > Note: When there's an Avatar and a Badge and the Avatar provided already has a Badge, the provided Badge will take over the provided Badge.
+- image
+
+There are 2 sizing variants:
+
+- stretch: When the text lines have a larger height compared to the coin, stretch the coin to fit text lines height.
+- fixed: Keep the coin the same size, no matter the height of the text lines.
+
+**⚠️Responsive text variants will be left out of the initial implementation due to the accessibility concerns, they are added to this spec to start a discussion⚠️**
+
+There are 4 responsive text variants:
+
+- truncate: simple text truncation when the text doesn't fit in the container.
+  - > Note: after a talk offline, it was brought up that this might be an accessibility concern. @smhighley could you confirm these scenarios are concerns for the text?
+- hidden: same as truncate, but has a default `min-width` of 36px. If the width is less than `min-width`, then the text is hidden.
+- wrap: simple text wrap.
+- lineclamp truncate: wraps text until specified number of lines and after those lines it truncates the text.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+**Slots**
+
+- `root`: The root slot for Persona.
+- `avatar`: The Avatar, if provided.
+- `badge`: The PresenceBadge, if provided.
+- `icon`: The icon, if provided.
+- `image`: The image, if provided.
```

Comment:
- Are these slots mutually exclusive? For example, can you use `<Persona image={foo} avatar={bar} />`?
- Are you going to bundle `Avatar` & `Image` components with `Persona`? I.e. will be there something like `state.components: { avatar: Avatar, image: Image }`?

Reply:
Looking at the design spec there is just a space called "coin slot" where you can put an avatar, a badge, an icon, or an image. So I think they are mutually exclusive which leads me to think we should have just a "coin slot" (need a better name imo) rather than `avatar`, `badge`, `icon` and `image`.

Reply:
There's a problem with this. If we were to add a single slot, we would need to either modify or add types for such as slot. To be able to have the single slot, its slot definition would need to look like this:

`media: NonNullable<Slot<typeof Avatar, typeof PresenceBadge | 'img' | 'span'>>;` 

This is not possible due to AlternateAs only allowing intrinsic elements. This would mean we could not have a badge alone.

Reply:
@spmonahan I was thinking about exactly the same, but wanted to check requirements first. Do you mean following ⬇️?

```tsx
function App() {
  return (
    <>
      <Persona media={<Avatar />} />
      <Persona media={<PresenceBadge />} />
      <Persona media={<Image />} />
      <Persona media={<SomeIcon />} />
    </>
  );
}
```

@sopranopillow are there reasons why following will not work?

Reply:
That's what I was thinking. @sopranopillow points out an issue with enforcing component types with the `Slot` type helper but I wonder if that really matters? Ultimately if we have `Slot<'span'>`, as in the example, anything can be slotted and it might be nice to reduce the slot to just a span to avoid the bundle size hit of importing Avatar, PresenceBadge and Image to use Persona.

I haven't thought this through but it seems we could also type the slot with:

```ts
media: NonNullable<Slot<typeof Avatar> | Slot<typeof PresenseBadge> | Slot<typeof Image> | Slot<'span'>>
```

Reply:
After trying out some options, I'll update the spec to have a generic `media` slot. Thanks for the feedback!